File: AllegroDotNet.Wrapper/src/AlFullscreen.cpp
#include "PCH.h"

#include "Al.h"
#include "AllegroDisplayMode.h"

namespace AllegroDotNet::Wrapper
{
	AllegroDisplayMode^ Al::GetDisplayMode(Int32 index, AllegroDisplayMode^ mode)
	{
		auto* nativeMode = mode->GetNativePointer();
		auto* nativeReturnValue = al_get_display_mode(index, nativeMode);
		return gcnew AllegroDisplayMode(nativeReturnValue);
	}

	Int32 Al::GetNumDisplayModes()
	{
		return al_get_num_display_modes();
	}
}

File: Source/AllegroDotNet/Al.Fullscreen.cs
using SubC.AllegroDotNet.Models;
using SubC.AllegroDotNet.Native;
using System.Runtime.InteropServices;

namespace SubC.AllegroDotNet;

/// <summary>
/// This static class contains the Allegro 5 library methods.
/// </summary>
public static partial class Al
{
    public static AllegroDisplayMode? GetDisplayMode(int index, ref AllegroDisplayMode mode)
    {
        var pointer = Interop.Core.AlGetDisplayMode(index, ref mode);
        return Marshal.PtrToStructure<AllegroDisplayMode>(pointer);
    }

    public static int GetNumDisplayModes()
    {
        return Interop.Core.AlGetNumDisplayModes();
    }
}
